src/components/create-from-builder/create-from-builder.controller.ts

```diff
@@ -0,0 +1,236 @@
+      name = _.kebabCase(name);
+
+      // Make sure it's a valid name before setting it.
+      if (this.ctrl.namePattern.test(name)) {
+        this.ctrl.name = name;
+      }
+    }
+  }
+
+  public createApp() {
+    this.createProjectIfNecessary().then(() => {
+      this.gotoStep(_.last(this.ctrl.steps));
+      // Get the image stream tag so we know what ports are exposed by the image.
+      this.getImageStreamTag().then((imageStreamTag: any) => {
+        let apiObjects = this.BuilderAppService.makeAPIObjects({
+          name: this.ctrl.name,
+          repository: this.ctrl.repository,
+          namespace: this.ctrl.selectedProject.metadata.name,
+          imageStreamTag: imageStreamTag
+        });
+        this.createAPIObjects(apiObjects);
+      }, (e: any) => {
+        this.ctrl.error = e;
+      });
+    }, (result) => {
+      let data = result.data || {};
+      if (data.reason === 'AlreadyExists') {
+        this.ctrl.projectNameTaken = true;
+      } else {
+        this.ctrl.error = data.message || 'An error occurred creating the project.';
+      }
+    });
+  };
+
+  public navigateToAdvancedForm() {
+    let template = 'project/{project}/create/fromimage?' +
+                     'imageStream={imageStream}&' +
+                     'imageTag={imageTag}&' +
+                     'namespace={namespace}&' +
+                     'displayName={displayName}&' +
+                     'name={name}&' +
+                     'sourceURI={sourceURI}&' +
+                     'advanced=true';
+    let target = URI.expand(template, {
+      project: this.ctrl.selectedProject.metadata.name,
+      imageStream: this.ctrl.imageStream.resource.metadata.name,
+      imageTag: this.ctrl.istag.name,
+      namespace: this.ctrl.imageStream.resource.metadata.namespace,
+      displayName: this.ctrl.imageStream.name,
+      name: this.ctrl.name || '',
+      sourceURI: this.ctrl.repository || ''
+    }).toString();
+
+    // TODO: Handle configurable base URLs.
+    // let base = 'https://localhost:9000/dev-console/';
+    // if (base) {
+    //   window.location.href = base + target;
+    //   return;
+    // }
+
+    this.$location.url(target);
+  }
+
+  private createProjectIfNecessary() {
+    if (_.has(this.ctrl.selectedProject, 'metadata.uid')) {
+      return this.$q.when();
+    }
+
+    // TODO: Common code from this controller and order service.
+    let newProjName = this.ctrl.selectedProject.metadata.name;
+    let newProjDisplayName = this.ctrl.selectedProject.metadata.annotations['new-display-name'];
+    let newProjDesc = this.$filter('description')(this.ctrl.selectedProject);
+    let projReqObj: any = {
+      apiVersion: "v1",
+      kind: "ProjectRequest",
+      metadata: {
+        name: newProjName
+      },
+      displayName: newProjDisplayName,
+      description: newProjDesc
+    };
+    return this.DataService.create('projectrequests', null, projReqObj, this.$scope);
```

Comment:
> I think this might have been resolved by openshift/origin#8587, although we should check with @liggitt or @deads2k

It's resolved in the common case.  There's still a time-based return which can cause a successful return even if no rolebindings have been observed.